client/allocrunner/alloc_runner_unix_test.go

```diff
@@ -154,9 +154,10 @@ func TestAllocRunner_Restore_CompletedBatch(t *testing.T) {
 	alloc := mock.Alloc()
 	alloc.Job.Type = structs.JobTypeBatch
 	task := alloc.Job.TaskGroups[0].Tasks[0]
-	task.Driver = "mock_driver"
+	task.Driver = "raw_exec"
 	task.Config = map[string]interface{}{
-		"run_for": "2ms",
+		"command": "sleep",
+		"args":    []string{"2"},
```

Comment:
I had to use the `raw_exec` driver instead of `mock_driver` here and in the other restore test (`TestTaskRunner_Restore_Dead`) because, for some reason, the `mock_driver` behaved differently from others.

For some reason the task handle was not `nil` when restoring a `dead` task, so the `Run` method would get stuck here, with `handleKill` never exiting: https://github.com/hashicorp/nomad/blob/4ac9de1aa619c852b31e02c809f5a4b30299a230/client/allocrunner/taskrunner/task_runner.go#L673-L674

I'm not sure if it's a problem with the `mock_driver` or my task restore changes.

Reply:
Probably worth getting to the bottom of this; just debug logging the plugin interface entrypoints and the parameters being passed in should shed light on what's happening

Reply:
Yup, thanks for the push back, my implementation was wrong 😅